cardinal/gamestate/map.go

```diff
@@ -18,257 +16,37 @@ func NewMapStorage[K comparable, V any]() *MapStorage[K, V] {
-	var res bool
-	var resAny any
-	resRaw, ok := m.internalMap[key]
-	if !ok {
-		return false, eris.New("does not exist in map")
-	}
-	resAny = resRaw
-	res, ok = resAny.(bool)
-	if !ok {
-		return false, eris.New("cannot convert to float")
-	}
-	return res, nil
-}
-func (m *MapStorage[K, V]) GetBytes(_ context.Context, key K) ([]byte, error) {
-	var res []byte
-	var resAny any
-	resRaw, ok := m.internalMap[key]
-	if !ok {
-		return nil, eris.New("does not exist in map")
+func (m *MapStorage[K, V]) Keys() ([]K, error) {
+	acc := make([]K, 0, len(m.internalMap))
+	for k := range m.internalMap {
+		acc = append(acc, k)
 	}
-	resAny = resRaw
-	res, ok = resAny.([]byte)
-	if !ok {
-		return nil, eris.New("cannot convert to float")
-	}
-	return res, nil
+	return acc, nil
 }
 
-func (m *MapStorage[K, V]) Set(_ context.Context, key K, value any) error {
-	typedValue, ok := value.(V)
-	if !ok {
-		return eris.New("could not type assert into type")
-	}
-	m.internalMap[key] = typedValue
+func (m *MapStorage[K, V]) Delete(key K) error {
+	delete(m.internalMap, key)
 	return nil
 }
 
-func incrementIfNumeric(value any) (any, bool) {
-	switch v := value.(type) {
-	case int:
-		return v + 1, true
-	case int8:
-		return v + 1, true
-	case int16:
-		return v + 1, true
-	case int32:
-		return v + 1, true
-	case int64:
-		return v + 1, true
-	case uint:
-		return v + 1, true
-	case uint8:
-		return v + 1, true
-	case uint16:
-		return v + 1, true
-	case uint32:
-		return v + 1, true
-	case uint64:
-		return v + 1, true
-	case float32:
-		return v + 1, true
-	case float64:
-		return v + 1, true
-	case complex64:
-		return v + complex(1, 0), true
-	case complex128:
-		return v + complex(1, 0), true
-	default:
-		return nil, false
-	}
-}
-
-func (m *MapStorage[K, V]) Incr(_ context.Context, key K) error {
-	value, ok := m.internalMap[key]
-	if !ok {
-		return eris.New("cannot find element to increment")
-	}
-	newValue, ok := incrementIfNumeric(value)
-	if !ok {
-		return eris.New("not numeric, cannot increment")
-	}
-	value, ok = newValue.(V)
+func (m *MapStorage[K, V]) Get(key K) (V, error) {
+	v, ok := m.internalMap[key]
 	if !ok {
-		return eris.New("cannot type assert back to original value")
-	}
-	m.internalMap[key] = value
-	return nil
-}
-
-func decrementIfNumeric(value any) (any, bool) {
-	switch v := value.(type) {
-	case int:
-		return v - 1, true
-	case int8:
-		return v - 1, true
-	case int16:
-		return v - 1, true
-	case int32:
-		return v - 1, true
-	case int64:
-		return v - 1, true
-	case uint:
-		return v - 1, true
-	case uint8:
-		return v - 1, true
-	case uint16:
-		return v - 1, true
-	case uint32:
-		return v - 1, true
-	case uint64:
-		return v - 1, true
-	case float32:
-		return v - 1, true
-	case float64:
-		return v - 1, true
-	case complex64:
-		return v - complex(1, 0), true
-	case complex128:
-		return v - complex(1, 0), true
-	default:
-		return nil, false
+		return v, eris.New("key not found")
```

Comment:
This interface should distinguish between actual errors and missing keys. "not found" could either be a dedicated error, or Get could return `(V, bool, error)`